src/Abstractions/NexusMods.Abstractions.GameLocators/WineParser.cs

```diff
@@ -0,0 +1,174 @@
+using System.Text;
+using JetBrains.Annotations;
+using NexusMods.Paths;
+
+namespace NexusMods.Abstractions.GameLocators;
+
+/// <summary>
+/// Parser for WINE.
+/// </summary>
+public static class WineParser
+{
+    public const string EnvironmentVariableName = "WINEDLLOVERRIDES";
+
+    /// <summary>
+    /// Parses the given `winetricks.log` file and returns all installed packages.
+    /// </summary>
+    public static IReadOnlySet<string> ParseWinetricksLogFile(AbsolutePath filePath)
+    {
+        if (!filePath.FileExists) return new HashSet<string>();
+
+        using var stream = filePath.Read();
+        return ParseWinetricksLogFile(stream);
+    }
+
+    /// <summary>
+    /// Parses the given `winetricks.log` file and returns all installed packages.
+    /// </summary>
+    public static IReadOnlySet<string> ParseWinetricksLogFile(Stream stream)
+    {
+        using var sr = new StreamReader(stream, Encoding.UTF8);
+
+        var result = new HashSet<string>();
+
+        while (sr.ReadLine() is { } line)
+        {
+            var trimmed = line.Trim();
+            if (trimmed.Length == 0) continue;
+
+            result.Add(line);
+        }
+
+        return result;
+    }
+
+    /// <summary>
+    /// Parses the <see cref="EnvironmentVariableName"/>.
+    /// </summary>
+    public static IReadOnlyList<WineDllOverride> ParseEnvironmentVariable(ReadOnlySpan<char> environmentVariableValue)
+    {
+        if (environmentVariableValue.Length == 0) return [];
+
+        var results = new List<WineDllOverride>();
+
+        // https://gitlab.winehq.org/wine/wine/-/wikis/Wine-User's-Guide#winedlloverrides-dll-overrides
+        var span = GetWineDllOverridesSection(environmentVariableValue);
+
+        // NOTE(erri120): DLLs are separated with a semicolon
+        var splitEnumerator = span.Split(';');
+        foreach (var splitRange in splitEnumerator)
+        {
+            var section = span[splitRange];
+
+            var index = section.LastIndexOf('=');
+            if (index == -1) continue;
+
+            var namesSpan = section[..index];
+
+            var dllNamesEnumerator = namesSpan.Split(',');
+            var dllOverrideTypes = GetOverrideTypes(section);
+
+            foreach (var range in dllNamesEnumerator)
+            {
+                var name = FixDllName(namesSpan[range]).ToString();
+                results.Add(new WineDllOverride(name, dllOverrideTypes));
+            }
+        }
+
+        return results;
+    }
+
+    private static ReadOnlySpan<char> FixDllName(ReadOnlySpan<char> input)
+    {
+        return input.TrimEnd(".dll");
```

Comment:
We need to ignore the case here!
This could be a `.DLL`, with DLL name in uppercase, have seen it before.

In other words, an IndexOf and Slice!